ksql-parser/src/main/java/io/confluent/ksql/util/ParserUtil.java

```diff
@@ -35,47 +38,51 @@
 import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 import org.antlr.v4.runtime.ParserRuleContext;
 import org.antlr.v4.runtime.Token;
 import org.antlr.v4.runtime.tree.TerminalNode;
-import org.apache.commons.lang3.ObjectUtils;
 
 public final class ParserUtil {
 
-  public static final Set<String> RESERVED_WORDS;
+  private ParserUtil() {
+  }
+
+  private static final Set<String> RESERVED_WORDS;
 
   static {
-    final ImmutableSet.Builder<String> reservedWords = new ImmutableSet.Builder<>();
-    for (int i = 1; i < SqlBaseParser.VOCABULARY.getMaxTokenType(); i++) {
-      final String name = ObjectUtils.firstNonNull(
-          SqlBaseParser.VOCABULARY.getLiteralName(i),
-          SqlBaseParser.VOCABULARY.getSymbolicName(i)
-      );
-      if (name == null) {
-        throw new IllegalStateException("Did not expect to unknown word in vocab at index " + i);
+    final KsqlParser parser = new DefaultKsqlParser();
+
+    final Predicate<String> isReservedWord = columnName -> {
+      try {
+        parser.parse(
+            "CREATE STREAM x (" + columnName + " INT) "
+                + "WITH(KAFKA_TOPIC='x', VALUE_FORMAT='JSON');");
+        return false;
+      } catch (final ParseFailedException e) {
+        return true;
       }
-      reservedWords.add(StringUtil.cleanQuotes(name));
-    }
-    RESERVED_WORDS = reservedWords.build();
+    };
+
+    final Set<String> reserved = IntStream.range(0, SqlBaseLexer.VOCABULARY.getMaxTokenType())
+        .mapToObj(SqlBaseLexer.VOCABULARY::getLiteralName)
+        .filter(Objects::nonNull)
+        .map(l -> l.substring(1, l.length() - 1)) // literals start and end with ' - remove them
+        .map(String::toUpperCase)
+        .filter(isReservedWord)
```

Comment:
haha nice - didn't think about doing it this way... can we add some unit tests for this though (words that are in vocabulary but are not reserved)?

Reply:
done.